icicle/backend/cpu/src/field/cpu_vec_ops.cpp

```diff
@@ -154,17 +155,17 @@ class VectorOpTask : public TaskBase
   // Single worker functionality to execute sum(vector)
   void vector_sum()
   {
-    *m_output = m_op_a[0];
-    for (uint64_t i = 1; i < m_nof_operations; ++i) {
-      *m_output = *m_output + m_op_a[i];
+    m_intermidiate_res = T::zero();
```

Comment:
why? was there a bug?

Reply:
yes. For reduction ops we should use m_intermidiate_res and not m_output. m_output will aggregate m_intermidiate_res from different threads.